### ISSUE TITLE
Fixing "solid/reactivity" rule in component props removes type annotation

### ISSUE THREAD

**Issue by micnic** [solidjs-community/eslint-plugin-solid#58]
**Describe the bug**
When calling fix issue for "solid/reactivity" in component props with type annotation the type annotation gets removed.

**To Reproduce**
```ts
type Props = {
  prop1: string;
  prop2: number;
};

const MyComponent = ({ prop1, prop2 }: Props) => { // Reactivity issue error
  return (
    <div>{prop1} {prop2}</div>
  );
};
```

**Expected behavior**
```ts
type Props = {
  prop1: string;
  prop2: number;
};

const MyComponent = (props: Props) => { // Type annotation should be kept
  return (
    <div>{props.prop1} {props.prop2}</div>
  );
};
```

**Actual behavior**

```ts
type Props = { // Types unused :(
  prop1: string;
  prop2: number;
};

const MyComponent = (props) => { // Type annotation is removed
  return (
    <div>{props.prop1} {props.prop2}</div>
  );
};
```

**Comment by joshwilsonvu** [comment]
Thanks so much for the quality bug report! Fixed in v0.9.3.

**Comment by micnic** [comment]
Thank you for fixing!
